pkg/virtual/framework/dynamic/apidefs/types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apidefs
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	apiresourcev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apiresource/v1alpha1"
+)
+
+// APIDomainKeyContextKeyType is the type of the key for the request context value
+// that will carry the API domain key.
+type APIDomainKeyContextKeyType string
+
+// APIDomainKeyContextKey is the key for the request context value
+// that will carry the API domain key.
+const APIDomainKeyContextKey APIDomainKeyContextKeyType = "VirtualWorkspaceAPIDomainLocationKey"
+
+// APIDefinition provides access to all the information needed to serve a given API resource
+type APIDefinition interface {
```

Comment:
What is an API? Are you talking about a resource? ResourceDefinition is a better fit?

Reply:
Still not convinced here, especially if we use a raw CRD instead of `CommonAPIResourceSpec`. I'd rather leave it for when  the refactoring mentioned here is done / validated.

Reply:
Am convinced that Resource is a better term :-) But as discussed we can wait until the APIBinding refactor.